controllers/machinehealthcheck_controller.go

```diff
@@ -274,7 +322,32 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 			t.string(),
 		)
 	}
+
 	for _, t := range healthy {
+		if m.Spec.RemediationTemplate != nil {
+			remediationObj := &corev1.ObjectReference{
+				APIVersion: m.Spec.RemediationTemplate.APIVersion,
+				Kind:       strings.TrimSuffix(m.Spec.RemediationTemplate.Kind, external.TemplateSuffix),
+				Name:       t.Machine.Name,
+			}
+
+			// Get remediation request object
+			obj, err := external.Get(ctx, r.Client, remediationObj, t.Machine.Namespace)
+			if err != nil {
+				if apierrors.IsNotFound(errors.Cause(err)) {
+					continue
+				}
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to fetch remediation request")
+			}
+			// Check that obj has no DeletionTimestamp to avoid hot loop
+			if obj.GetDeletionTimestamp() == nil {
+				// Issue a delete for remediation request.
+				if err := r.Client.Delete(ctx, obj); err != nil && !apierrors.IsNotFound(err) {
+					return ctrl.Result{}, errors.Wrapf(err, "Failed to delete %v %q for Machine %q", obj.GroupVersionKind(), obj.GetName(), t.Machine.Name)
```

Comment:
The first letter of an error message needs to be lowercase.

Do we surface this error in a condition anywhere? We need to make sure the log is not the sole place it goes.